From 32b8272accefcb98c3c4411718323092fec313af Mon Sep 17 00:00:00 2001
From: James Xiong <james.xiong@intel.com>
Date: Wed, 16 Sep 2020 12:48:59 -0700
Subject: [PATCH 1/1] drm/i915/wa: wait for hardware context state to be
 unpinned

there seems to be a race condition that the context state was
still pinned when it's unbinding, the problem is random and very
time sensitive, only happens with some particular releases, adding
a wait to make sure the state is unpinned as a workaround.

please note this is a w/a for 5.4.53 kernel tree and should not be
ported forward.

Signed-off-by: James Xiong <james.xiong@intel.com>
---
 drivers/gpu/drm/i915/i915_gem.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 927559cbb..9895bdcdd 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -1139,6 +1139,9 @@ static int __intel_engines_record_defaults(struct intel_gt *gt)
 		if (!state)
 			continue;
 
+		if(err = wait_for(!i915_vma_is_pinned(state), 50))
+			goto out;
+
 		/*
 		 * As we will hold a reference to the logical state, it will
 		 * not be torn down with the context, and importantly the
-- 
2.17.1

